guides/docs/testing/testing_controllers.md

```diff
@@ -133,169 +222,489 @@ defmodule HelloWeb.Router do
+When we run the test again, our failing test tells us module `HelloWeb.UserView` is not available. Let's add it. Our test specifies a JSON format with a top key of `"data"`, containing an array of users with attributes `"name"` and `"email"`.
 
 ```elixir
+# lib/hello_web/views/user_view.ex
+
 defmodule HelloWeb.UserView do
   use HelloWeb, :view
 
-  def render("index.json", %{users: users}) do
-    %{data: render_many(users, HelloWeb.UserView, "user.json")}
+  def render("index.json", %{data: users}) do
+    %{data:
+      render_many( users, HelloWeb.UserView, "user.json", as: :data)
+      }
   end
 
-  def render("user.json", %{user: user}) do
-    %{name: user.name, email: user.email}
+  def render("user.json", %{data: user}) do
+    %{
+      name: user.name,
+      email: user.email
+      # skipping password, inserted_at, and updated_at
+    }
   end
-
 end
 ```
 
+The view module for the index uses the `render_many/4` function.
+According to the
+[documentation](https://hexdocs.pm/phoenix/Phoenix.View.html#render_many/4),
+using `render_many/4` is "roughly equivalent" to using `Enum.map/2`,
+and in fact `Enum.map` is called under the hood.  The main difference
+is that by using `render_many/4` instead of directly calling
+`Enum.map/2` is that the former benefits from library-quality error
+checking, properly handling missing values, and so on.
+
 And with that, our test passes when we run it.
 
+### Time for the show
+
 We'll also cover the `show/2` action here so we can see how to handle an error case.
 
 Our show tests currently look like this:
 
 ```elixir
   describe "show/2" do
-    test "Responds with a newly created user if the user is found"
+    test "Responds with user info if the user is found"
     test "Responds with a message indicating user not found"
   end
 ```
 
-Run this test only by running the following command: (if your show tests don't start on line 32, change the line number accordingly)
+Run this test only by running the following command: (if your show tests don't start on line 41, change the line number accordingly)
 
 ```console
-$ mix test test/hello_web/controllers/user_controller_test.exs:32
+$ mix test test/hello_web/controllers/user_controller_test.exs:41
 ```
 
 Our first `show/2` test result is, as expected, not implemented.
 Let's build a test around what we think a successful `show/2` should look like.
 
 ```elixir
-test "Responds with a newly created user if the user is found" do
-  user = User.changeset(%User{}, %{name: "John", email: "john@example.com"})
-  |> Repo.insert!
+# test/hello_phoenix_web/controllers/user_controller_test.exs line 41 (or so)
 
-  response = build_conn()
-  |> get(user_path(build_conn(), :show, user.id))
+test "Responds with user info if the user is found", %{conn: conn, user1: user} do
```

Comment:
To create the user here, we should use:

    setup [:create_user]

Inside the describe and create a private `create_user` function. This is how the generators work for `gen.html` and `gen.json` https://github.com/phoenixframework/phoenix/blob/ede64f542e201934b36e6c8c5c9e8bd7046242ee/priv/templates/phx.gen.json/controller_test.exs

We should also mention how describe works. I would say something llike:

 > We use the `setup` macro in the `describe` to call a function before running each test. In this case, we want to create a user that we will display for the show action. The functions called by `setup` take an [ExUnit context](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#module-context) (not to be confused with the contexts we are describing throughout this guide) and allow us to add additional fields when we return.

Reply:
Sure.  Again, I just figured out how all this transaction wrapping stuff works today.

Reply:
addressed...